### DISCUSSION TITLE
mosquitto_pub failed over 1883

### DISCUSSION THREAD

**Discussion by arvindsingh-dev** [emqx/emqx#8809] [Q&A]
I am facing a weird issue after spinning up a local EMQX Docker container via **docker run -d --name emqx -p 1883:1883 -p 8083:8083 -p 8084:8084 -p 8883:8883 -p 18083:18083 emqx/emqx:latest**
also setting redis Authentiocation from Dashboard UI. 

While ws connection from MQTTX and WebSocket Client from Dashboard, but when trying connecting port 1883, its not getting authorized 
mosquitto_pub -h localhost -p 1883 -u user -P user -t "testtopic/1" -i  test2 -m  "{'msg': 'hello'}" -d
_Client test2 sending CONNECT
Client test2 received CONNACK (5)
Connection error: Connection Refused: not authorised.
Error: The connection was refused._

Is there something I am missing here? nothing specific when checking log file.

**Comment by zmstone** [comment]
Hi @arvindsingh-dev 

Do you see any logs from EMQX ?
If not, on the dashboard, you can go to Diagnose -> Log Trace to trace the client.
here is an example (I used built-in database)
<img width="1646" alt="image" src="https://user-images.githubusercontent.com/164324/186931793-7a64b05b-59b1-4b6c-aac0-156b8271e421.png">

**Reply by arvindsingh-dev** [reply]
Thanks  @zmstone for trace option.  I tried that but nothing specific was there. it turned out to be redis DB salt position of ( prefix, suffix) while setting up authentication. after that correction Authentication is working. but redis ACL not working. 

Any further help is much appreciated. 

i have setup 2 users i.e. **user123** and **user1234**

**User123**  with with all permission on testtopic/1
![image](https://user-images.githubusercontent.com/110582808/187157142-da6ce9c6-a831-4798-9459-51135bb1c671.png)

**User1234** with only subscribe permission on testtopic/1 

![image](https://user-images.githubusercontent.com/110582808/187157099-eebf861c-afd3-4d5d-8f4d-beff3837fbdc.png)

even after these setup correctly and in log as well its **User1234** is unauthorized after acl query still **User1234** is able to do subscribe/publish topic **testtopic/1** and **user123** can receive them
refer logs of user1234 [emqx_Mjg2ND.zip](https://github.com/emqx/emqx/files/9442898/emqx_Mjg2ND.zip)
refer logs of user123 [mqtt_testing_client.zip](https://github.com/emqx/emqx/files/9442899/mqtt_testing_client.zip)
